tools/parity_scale_decoder/src/main.rs

```diff
@@ -0,0 +1,279 @@
+        DataEntityFilter,
+        DataEvent,
+        DataEventFilter,
+        Divide,
+        Domain,
+        DomainEvent,
+        DomainEventFilter,
+        DomainFilter,
+        DomainId,
+        Equal,
+        Executable,
+        ExecuteTriggerBox,
+        ExecuteTriggerEvent,
+        ExecuteTriggerEventFilter,
+        ExecutionTime,
+        Expression,
+        FailBox,
+        FindAccountById,
+        FindAccountKeyValueByIdAndKey,
+        FindAccountsByDomainId,
+        FindAccountsByName,
+        FindAllAccounts,
+        FindAllAssets,
+        FindAllAssetsDefinitions,
+        FindAllDomains,
+        FindAllParameters,
+        FindAllPeers,
+        FindAllRoles,
+        FindAssetById,
+        FindAssetDefinitionKeyValueByIdAndKey,
+        FindAssetKeyValueByIdAndKey,
+        FindAssetQuantityById,
+        FindAssetsByAccountId,
+        FindAssetsByAssetDefinitionId,
+        FindAssetsByDomainId,
+        FindAssetsByDomainIdAndAssetDefinitionId,
+        FindAssetsByName,
+        FindDomainById,
+        FindDomainKeyValueByIdAndKey,
+        FindPermissionTokensByAccountId,
+        FindRolesByAccountId,
+        FindTransactionByHash,
+        FindTransactionsByAccountId,
+        GenesisDomain,
+        GrantBox,
+        Greater,
+        IfExpression,
+        IfInstruction,
+        Instruction,
+        InstructionExecutionFail,
+        Less,
+        Metadata,
+        MetadataLimits,
+        MintBox,
+        Mod,
+        Multiply,
+        Not,
+        NotPermittedFail,
+        Or,
+        Pair,
+        Payload,
+        Peer,
+        PeerEvent,
+        PeerEventFilter,
+        PeerFilter,
+        PeerId,
+        PendingTransactions,
+        PermissionToken,
+        PipelineEntityKind,
+        PipelineEvent,
+        PipelineEventFilter,
+        PipelineStatus,
+        QueryBox,
+        QueryRequest,
+        QueryResult,
+        RaiseTo,
+        RegisterBox,
+        RejectedTransaction,
+        RejectionReason,
+        RemoveKeyValueBox,
+        Repeats,
+        RevokeBox,
+        Role,
+        RoleEvent,
+        RoleEventFilter,
+        RoleFilter,
+        RoleId,
+        SequenceBox,
+        SetKeyValueBox,
+        SignatureCheckCondition,
+        SignedQueryRequest,
+        Subtract,
+        TimeEvent,
+        TimeEventFilter,
+        TimeInterval,
+        TimeSchedule,
+        Transaction,
+        TransactionRejectionReason,
+        TransactionValue,
+        TransferBox,
+        Trigger,
+        TriggerEvent,
+        TriggerEventFilter,
+        TriggerFilter,
+        TriggerId,
+        UnregisterBox,
+        UnsatisfiedSignatureConditionFail,
+        ValidTransaction,
+        VersionedPendingTransactions,
+        VersionedQueryResult,
+        VersionedRejectedTransaction,
+        VersionedSignedQueryRequest,
+        VersionedTransaction,
+        VersionedValidTransaction,
+        WasmExecutionFail,
+        Where,
+    )
```

Comment:
as you're aware types have to be manually added here and it can be easily forgotten to do so. To mitigate this concern as much as possible I would like if you would make a test which would use `build_schemas` from `iroha_schema` and make sure that at least those types are included here. I don't think there is much we can do for internal types, we'll just have to be careful about listing them here

Reply:
so at least with a simple test we can guarantee that types listed in `build_schemas` are a subset of types listed here

Reply:
Good idea.
I was originally going to vote for the 1st implementation, but provided the drawback is somehow mitigated I'd prefer this 2nd one

Reply:
Not all types, that implements `IntoSchema` also implements `Decode`. So I've made a test with a few exceptions